### PR TITLE
Helm chart: Support providing RollingUpdate maxSurge and maxUnavailab…

### LINKED ISSUE #5664: Helm chart: Support providing RollingUpdate maxSurge and maxUnavailable values
## Is your feature request related to a problem? Please describe.
RollingUpdate maxSurge and maxUnavailable are commonly used configuration values, but are not currently available to be be set from the router helm chart.  

## Describe the solution you'd like
Make these values configurable from the router helm chart.

## Describe alternatives you've considered
Not using the router helm chart

## Additional context
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment

### PR DESCRIPTION
**Pull request by theJC**
…le values

RollingUpdate maxSurge and maxUnavailable are commonly used deployment configuration values.  This change makes their values able to be set via the router helm chart. 

Fixes #5664

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: helm/chart/router/templates/deployment.yaml [hunk 1]
```diff
@@ -172,3 +172,13 @@ spec:
...
+  {{- if .Values.rollingUpdate }}
+  strategy:
+    rollingUpdate:
+      {{- if  (hasKey .Values.rollingUpdate "maxUnavailable") }}
+      maxUnavailable: {{ .Values.rollingUpdate.maxUnavailable }}
+      {{- end }}
+      {{- if (hasKey .Values.rollingUpdate "maxSurge") }}
+      maxSurge: {{ .Values.rollingUpdate.maxSurge }}
+      {{- end }}
+  {{- end }}
```
Comment (cyberhck) [inline helm/chart/router/templates/deployment.yaml:184]:
```suggestion
  {{- end }}

```

**Review by BrynCooke — CHANGES_REQUESTED** [review]
Looks good but needs `helm/chart/router/README.md` updating.

File: helm/chart/router/values.yaml [hunk 2]
```diff
@@ -218,6 +218,15 @@ autoscaling:
...
 
+
+rollingUpdate:
+  {}
+# Adjust rolling update strategy.  Can take absolute values or % values.
+# (https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment)
+# Defaults if not set are:
+#  maxUnavailable: 25%
+#  maxSurge: 25%
+
```
Comment (garypen) [inline helm/chart/router/values.yaml:229]: Would be nice to have line 224 moved to line 221 and specified as a helm-docs style comment

Comment (theJC) [reply helm/chart/router/values.yaml]: I've adjusted and used helm-doc style comment

Comment (theJC) [reply helm/chart/router/values.yaml]: @garypen - are you good with this PR now with the adjustment made?

Comment (theJC) [reply helm/chart/router/values.yaml]: @garypen -- Sorry for the harassment, but can we get this merged in.  I have at least one more PR I'd like to submit, but I'd like this one settled first before I do so.

Comment (garypen) [reply helm/chart/router/values.yaml]: @theJC Sorry about the delay, I've been on vacation.

**Review by garypen — APPROVED** [review]

**Pull request merged**